controllers/kserve_customcacert_controller.go

```diff
@@ -0,0 +1,190 @@
+)
+
+type KServeCustomCACertReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	Log    logr.Logger
+}
+
+// reconcileConfigMap watch odh global ca cert and it will create/update/delete kserve custom cert configmap
+func (r *KServeCustomCACertReconciler) reconcileConfigMap(configmap *corev1.ConfigMap, ctx context.Context) error {
+	// Initialize logger format
+	log := r.Log
+
+	odhCustomCertData := configmap.Data[constants.ODHCustomCACertFileName]
+	if odhCustomCertData == "" {
+		log.Info(fmt.Sprintf("Detected opendatahub global cert ConfigMap (%s), but custom cert is not set\n", odhGlobalCACertConfigMapName))
+		kserveCustomCertConfigMap := &corev1.ConfigMap{}
+		err := r.Get(ctx, types.NamespacedName{Name: kserveCustomCACertConfigMapName, Namespace: configmap.Namespace}, kserveCustomCertConfigMap)
+		if err == nil {
+			log.Info(fmt.Sprintf("Deleting KServe custom cert ConfigMap (%s)", kserveCustomCACertConfigMapName))
+			if err := r.Delete(ctx, kserveCustomCertConfigMap); err != nil {
+				return err
+			}
+		} else if !apierrs.IsNotFound(err) {
+			return err
+		}
+		return nil
+	}
+
+	configData := map[string]string{constants.KServeCACertFileName: odhCustomCertData}
+	newCaCertConfigMap := getDesiredCaCertConfigMapForKServe(kserveCustomCACertConfigMapName, configmap.Namespace, configData)
+
+	kserveCustomCertConfigMap := &corev1.ConfigMap{}
+	if err := r.Get(ctx, types.NamespacedName{Name: kserveCustomCACertConfigMapName, Namespace: configmap.Namespace}, kserveCustomCertConfigMap); err != nil {
+		if !apierrs.IsNotFound(err) {
+			return err
+		}
+		log.Info(fmt.Sprintf("Creating KServe custom cert ConfigMap (%s) because it detected the creation of the opendatahub global cert ConfigMap (%s)", kserveCustomCACertConfigMapName, odhGlobalCACertConfigMapName))
+		if err := r.Create(ctx, newCaCertConfigMap); err != nil {
+			log.Error(err, "Failed to create KServe custom cert ConfigMap")
+			return err
+		}
+	} else {
+		log.Info("Checking if the data in KServe custom cert ConfigMap differs from the data in the opendatahub global CA cert ConfigMap")
+		existingOdhCustomCertData := kserveCustomCertConfigMap.Data[constants.KServeCACertFileName]
+		if existingOdhCustomCertData == odhCustomCertData {
+			log.Info(fmt.Sprintf("No updates required for KServe custom cert ConfigMap (%s) as the data matches the opendatahub global cert ConfigMap (%s)", kserveCustomCACertConfigMapName, odhGlobalCACertConfigMapName))
+			return nil
+		}
+		log.Info(fmt.Sprintf("Updating KServe custom cert ConfigMap due to changes in the opendatahub global cert ConfigMap (%s)", odhGlobalCACertConfigMapName))
+		if err := r.Update(ctx, newCaCertConfigMap); err != nil {
+			return err
+		}
+		if err := r.deleteStorageSecret(ctx, configmap.Namespace); err != nil {
+			log.Error(err, "Failed to delete the storage-config secret to update the custom cert")
+			return err
+		}
+		log.V(1).Info("Deleted the storage-config Secret to update the custom cert")
```

Comment:
I agreed. What do you think if we delete the storage-config secret when it has `"opendatahub.io/managed: "true"` label?  That means the secret is managed by odh-model-controller.

Reply:
Does `storage-config` secret created by `odh-model-controller` in any case? 
So far the way I tested the KServe is by manually create the `storage-config` secret in Kserve namespace.

If there are some cases where `storage-config` secret created by `odh-model-controller` then I am agree with you to delete them by checking on `"opendatahub.io/managed: "true"` label.

Reply:
Yes, it is.
`storage-config` can be manually created but in ODH/RHOAI, it will be managed by data connection and `odh-model-controller` creates the secret.